### PR TITLE
Fullscreen Settings for Reader

### PR DESCRIPTION
**Pull request by jobobby04**
- [Add a setting to disable fullscreen](https://i.imgur.com/yrmCRyc.gif)
- [Add a setting to match fullscreen to focus mode](https://i.imgur.com/mFDCDqz.gif)

Depends on #236 due to the settings composables

### CODE REVIEW

File: android/src/main/java/app/shosetsu/android/ui/reader/ChapterReader.kt [hunk 1]
```diff
@@ -249,13 +250,14 @@ class ChapterReader
 								//item { viewModel.horizontalSwitchOption() }
 								item { viewModel.invertChapterSwipeOption() }
 								item { viewModel.readerKeepScreenOnOption() }
+								item { viewModel.enableFullscreen() }
 								item { viewModel.showReaderDivider() }
 								item { viewModel.stringAsHtmlOption() }
 								item { viewModel.doubleTapFocus() }
-								item { viewModel.doubleTapSystem() }
+								item { viewModel.doubleTapSystem(enableFullscren) }
```
Comment (Doomsdayrs) [inline android/src/main/java/app/shosetsu/android/ui/reader/ChapterReader.kt]: The premise of the viewnodel extension here is removal of redundant boiler plate for observing the view model. Move observation into the setting item itself.

File: android/src/main/java/app/shosetsu/android/ui/settings/sub/ReaderSettings.kt [hunk 2]
```diff
@@ -314,12 +315,17 @@ fun ReaderSettingsContent(
...
+		item { viewModel.enableFullscreen() }
+
 		item {
 			viewModel.showReaderDivider()
 		}
 
 		item { viewModel.doubleTapFocus() }
-		item { viewModel.doubleTapSystem() }
+		item {
+			val enableFullscreen by viewModel.enableFullscreen.collectAsState(true)
```
Comment (Doomsdayrs) [inline android/src/main/java/app/shosetsu/android/ui/settings/sub/ReaderSettings.kt]: Refer to previous boiler plate note

File: android/src/main/java/app/shosetsu/android/ui/reader/ChapterReader.kt (hunk 1 above)
Comment (jobobby04) [reply android/src/main/java/app/shosetsu/android/ui/reader/ChapterReader.kt]: 👍

File: android/src/main/java/app/shosetsu/android/ui/settings/sub/ReaderSettings.kt (hunk 2 above)
Comment (jobobby04) [reply android/src/main/java/app/shosetsu/android/ui/settings/sub/ReaderSettings.kt]: 👍

**Comment by jobobby04** [comment]
Rebased and ready

**Review by Doomsdayrs — APPROVED** [review]
I will merge this after testing it

**Comment by Doomsdayrs** [comment]
1. When entering a chapter, full screen is not the first thing that occurs, instead one needs to focus to have full screen kick in
2. I can still click the toggle the disabled settings (double tap full screen and match full screen)

**Review by Doomsdayrs — CHANGES_REQUESTED** [review]
1. When entering a chapter, full screen is not the first thing that occurs, instead one needs to focus to have full screen kick in
2. I can still click the toggle the disabled settings (double tap full screen and match full screen)

**Comment by jobobby04** [comment]
1. Thats the current behavior I beleive? I tested with the latest preview and that currently is what happens.
2. Fixed

**Comment by Doomsdayrs** [comment]
> 1. Thats the current behavior I believe? I tested with the latest preview and that currently is what happens.
> 
>     2. Fixed

1. Huh, I never noticed it, nvm then
2. Nice

**Review by Doomsdayrs — APPROVED** [review]

**Pull request merged**